### PR TITLE
docs: Add sklearnex spmd documentation

### PR DESCRIPTION
**Pull request by ethanglaser**
### Description

Add docs for scikit-learn-intelex spmd

image

_List issue number(s) if exist(s): #6 (for example)_

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

File: doc/sources/algorithms.rst [hunk 1]
```diff
@@ -362,6 +364,174 @@ Nearest Neighbors
        - ``metric`` not in [`'euclidean'`, `'manhattan'`, `'minkowski'`, `'chebyshev'`, `'cosine'`]
      - Only dense data is supported
 
+Other tasks
```
Comment (bdmoore1) [inline doc/sources/algorithms.rst:367]:
```suggestion
Other Tasks
```

File: doc/sources/algorithms.rst [hunk 2]
```diff
@@ -362,6 +364,174 @@ Nearest Neighbors
...
+       - ``Init`` = `'k-means++'` fallbacks to CPU.
+     - Sparse data is not supported
+   * - `DBSCAN`
+     - All parameters are supported except:
+
+       - ``metric`` != `'euclidean'`
+       - ``algorithm`` not in [`'brute'`, `'auto'`]
+     - Only dense data is supported
+
+Dimensionality reduction
```
Comment (bdmoore1) [inline doc/sources/algorithms.rst:483]:
```suggestion
Dimensionality Reduction
```

File: doc/sources/algorithms.rst [hunk 3]
```diff
@@ -362,6 +364,174 @@ Nearest Neighbors
...
+     - Data formats
+   * - `NearestNeighbors`
+     - All parameters are supported except:
+
+       - ``algorithm`` != `'brute'`
+       - ``weights`` = `'callable'`
+       - ``metric`` not in [`'euclidean'`, `'manhattan'`, `'minkowski'`, `'chebyshev'`, `'cosine'`]
+     - Only dense data is supported
+
+Other tasks
```
Comment (bdmoore1) [inline doc/sources/algorithms.rst:520]:
```suggestion
Other Tasks
```

Comment (bdmoore1) [inline doc/sources/algorithms.rst:520]: There are two sections titled "Other Tasks". Is that intentional?

File: doc/sources/distributed-mode.rst [hunk 4]
```diff
@@ -19,12 +19,26 @@
...
+example of this process in the |intelex| repository, where many examples of our SPMD-supported estimators are 
+available: https://github.com/intel/scikit-learn-intelex/blob/main/examples/sklearnex/. To run:
 
-While daal4py is available in `distribued mode <https://intelpython.github.io/daal4py/scaling.html>`_,
-|intelex| does not currently offer this functionality.
+::
+
+  mpirun -n 4 python linear_regression_spmd.py
+
+Note that additional mpirun arguments can be added as desired. SPMD-supported estimators are listed below
```
Comment (bdmoore1) [inline doc/sources/distributed-mode.rst:40]:
```suggestion
Note that additional mpirun arguments can be added as desired. SPMD-supported estimators are listed in the
```

**Review by bdmoore1 — APPROVED** [review]

File: doc/sources/algorithms.rst [hunk 5]
```diff
@@ -362,6 +364,174 @@ Nearest Neighbors
...
+   :align: left
+
+   * - Algorithm
+     - Parameters & Methods
+     - Data formats
+   * - `RandomForestRegressor`
+     - All parameters are supported except:
+
+       - ``warm_start`` = `True`
+       - ``cpp_alpha`` != `0`
```
Comment (md-shafiul-alam) [inline doc/sources/algorithms.rst:439]:
```suggestion
       - ``ccp_alpha`` != `0`
```

Comment (md-shafiul-alam) [reply doc/sources/algorithms.rst:439]: A couple of other places also have this typo.

File: doc/sources/algorithms.rst (hunk 3 above)
Comment (ethanglaser) [reply doc/sources/algorithms.rst:520]: Yes, there are the same subsections for each different hardware (CPU, GPU, SPMD)

**Review by md-shafiul-alam — APPROVED** [review]
